kpm/lib/kpm/system_helpers/cpu_information.rb

```diff
@@ -0,0 +1,70 @@
+module SystemProxy
+  module CpuInformation
+    class << self
+      def fetch
+        cpu_info = nil
+        if OS.windows?
+          cpu_info = fetch_windows
+        elsif OS.linux?
+          cpu_info = fetch_linux
+        elsif OS.mac?
+          cpu_info = fetch_mac
+        end
+
+        cpu_info
+      end
+
+      def get_labels
+        labels = [{:label => :cpu_detail},
+                  {:label => :value}]
+        labels
+      end
+
+      private
+        def fetch_linux
+          cpu_data = `cat /proc/cpuinfo 2>&1`.gsub("\t",'')
+          cpu = get_hash(cpu_data)
+          cpu
+        end
+
+        def fetch_mac
+          cpu_data = `system_profiler | grep -e "Processor Name" -e "Processor Speed" -e "Number of Processors" 2>&1`
```

Comment:
Running the full `system_profiler` is quite slow, but it looks like you only need `SPHardwareDataType` (i.e. use `system_profiler SPHardwareDataType` instead). Same below.

Could you also return the number of cores (`Total Number of Cores`)?